processor/attributesprocessor/attributes_metric_test.go

```diff
@@ -87,8 +88,14 @@ func generateMetricData(resourceName string, attrs map[string]pdata.AttributeVal
 			pdata.NewAttributeMapFromMap(attrs).CopyTo(dps.At(i).Attributes())
 			dps.At(i).Attributes().Sort()
 		}
+	default:
```

Comment:
Tbqh this entire switch block is redundant but I'm leaving it in for now  to reduce churn on the pr

Reply:
I am not sure why it is needed. Also, removing this block and uncommenting the [test cases where you found issues](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/7930/commits/94e2bf78b03c18cb826238909b8ce8bc8775bcf1#r825433525) leads to green tests :)

Reply:
That's because the tests don't actually test anything. If you change the expected attributes randomly, too, they also pass